File: leetcode_75/1_Array_String/cpp/1431.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution
{
public:
    vector<bool> kidsWithCandies(vector<int> &candies, int extraCandies)
    {
        int base_mx = 0;
        for (auto c : candies)
        {
            base_mx = max(base_mx, c);
        }

        vector<bool> result(candies.size());
        for (int i = 0; i < candies.size(); ++i)
        {
            result[i] = candies[i] + extraCandies >= base_mx ? true : false;
        }

        return result;
    }
};

File: leetcode_75/1_Array_String/python/1431.py
from typing import List


class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        base_mx = max(candies)
        result = list(map(lambda x: x + extraCandies >= base_mx, candies))
        return result
